### PR TITLE
fixing joblib version to 1.2

### PR DESCRIPTION
**Pull request by vuillaut**
Fixing #1056

### CODE REVIEW

**Review by maxnoe — CHANGES_REQUESTED** [review]
@vuillaut No! This is the wrong solution. There are files already produced with joblib 1.2, which are not readable with 1.1. So this would force everyone to use 1.1, which is incompatible with some files produced.

The correct solution is to pin 1.2

File: environment.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ dependencies:
   - matplotlib=3.5
   - notebook
   - iminuit>=2
-  - joblib
+  - joblib~=1.1
```
Comment (maxnoe) [inline environment.yml:18]: This means ">=1.1,<2.0"  so it does not in fact pin to 1.1

File: environment.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ dependencies:
   - matplotlib=3.5
   - notebook
   - iminuit>=2
-  - joblib
+  - joblib==1.2
```
Comment (maxnoe) [inline environment.yml:18]: I think this will pin to `1.2.0`, we should pin to `1.2.x` I think. `joblib=1.2` here

File: setup.py [hunk 3]
```diff
@@ -48,7 +48,7 @@ def find_scripts(script_dir, prefix):
         'gammapy~=0.19.0',
         'h5py',
         'iminuit>=2',
-        'joblib',
+        'joblib==1.2',
```
Comment (maxnoe) [inline setup.py:51]: and `~=1.2.0` here

File: environment.yml (hunk 2 above)
Comment (vuillaut) [reply environment.yml:18]: does it? 🤔 
Damn, ok changing

**Review by maxnoe — APPROVED** [review]

**Pull request merged**